src/Clients/GuzzleClient.php

```diff
@@ -18,47 +20,99 @@ class GuzzleClient implements Client
      */
     const DEFAULT_METHOD = 'GET';
 
+    /**
+     * Supported HTTP methods
+     */
+    const SUPPORTED_METHODS = [
+        'GET' => true,
+        'HEAD' => true,
+        'POST' => true,
+        'PUT' => true,
+        'DELETE' => true,
+        'OPTIONS' => true,
+    ];
+
     /**
      * @var array the Client main options
      */
     private $mainOptions;
 
+
     public function __construct(array $mainOptions = [])
     {
         $this->mainOptions = $mainOptions;
     }
 
     /**
      * {@inheritdoc}
+     * @throws UnavailableServiceException
      */
     public function request($resource, array $options)
     {
         try {
-            $client = new OriginalGuzzleClient($this->mainOptions);
-            $method = $this->defineMethod($options);
-            $options['http_errors'] = true;
+            $client = $this->buildClient();
+            $method = $this->getHttpMethod($options);
+            $options['exceptions'] = true;
+
+            // prevents unhandled method errors in Guzzle 5
+            unset($options['method']);
 
-            return (string) $client->request($method, $resource, $options)->getBody();
-        } catch (Exception $exception) {
-            throw new UnavailableService($exception->getMessage());
+            $request = $client->createRequest($method, $resource, $options);
+            return (string) $client->send($request)->getBody();
+        } catch (Exception $e) {
+            throw new UnavailableServiceException($e->getMessage(), $e->getCode(), $e);
         }
     }
 
     /**
      * @param array $options the list of options
      *
      * @return string the method
+     * @throws UnsupportedMethodException
      */
-    private function defineMethod(array $options)
+    private function getHttpMethod(array $options)
     {
         if (isset($this->mainOptions['method'])) {
             return $this->mainOptions['method'];
         }
 
         if (isset($options['method'])) {
+            if (!array_key_exists($options['method'], self::SUPPORTED_METHODS)
```

Comment:
```suggestion
            if (empty(self::SUPPORTED_METHODS[$options['method']])) {
)
```

Reply:
That would throw a notice in PHP < 7:

See https://3v4l.org/G76MH

Reply:
It sucks!